torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -201,5 +201,27 @@ Tensor SoftMarginLossImpl::forward(const Tensor& input, const Tensor& target) {
   return F::soft_margin_loss(input, target, options);
 }
 
+// ============================================================================
+
+NLLLossImpl::NLLLossImpl(
+    const NLLLossOptions& options_)
+    : options(options_) {
+    reset();
+}
+
+void NLLLossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::NLLLoss()";
+}
+
+void NLLLossImpl::reset() {
+  register_buffer("weight", options.weight());
```

Comment:
I think we can write
```suggestion
  weight = register_buffer("weight", options.weight());
```